src/pandablocks_ioc/_hdf_ioc.py

```diff
@@ -532,6 +624,12 @@ async def _handle_hdf5_data(self) -> None:
         This method expects to be run as an asyncio Task."""
         try:
             # Set up the hdf buffer
+
+            if not self._directory_exists_record.get() == 1:
+                raise RuntimeError(
+                    "Configured HDF directory does not exist or is not writable!"
+                )
```

Comment:
Should this condition have a test explicitly for it? I note there are tests where it must be `1` or they break, but I'd like to see one that does expect the break to occur.

Reply:
Added test to account for this.